test/test_pipeline_yaml.py

```diff
@@ -792,3 +847,45 @@ def test_save_yaml_overwrite(tmp_path):
     with open(tmp_path / "saved_pipeline.yml", "r") as saved_yaml:
         content = saved_yaml.read()
         assert content != ""
+
+
+def test_save_yaml_with_custom_node_kwargs(tmp_path):
+    class CustomNode(MockNode):
+        def __init__(self, param: int):
+            super().__init__()
+            self.param = param
+
+    pipeline = Pipeline()
+    pipeline.add_node(CustomNode(param=1), name="custom_node", inputs=["Query"])
+    pipeline.save_to_yaml(tmp_path / "saved_pipeline.yml")
+
+    print(pipeline.get_node("custom_node")._component_config)
+
+    with open(tmp_path / "saved_pipeline.yml", "r") as saved_yaml:
+        content = saved_yaml.read()
+
+        assert content.count("custom_node") == 2
+        assert "CustomNode" in content
+        assert "param: 1" in content
+        assert "Query" in content
+        assert f"version: {haystack.__version__}" in content
```

Comment:
This is a bit hard to understand and not really precise as "param: 1" could be part of any entry within the yaml. I'd prefer to check the dict returned by `pipeline.get_config()`